src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,11 +25,17 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
```

Comment:
good that you've removed the value - you can always reconstruct this from the individual component objects